Issue #399: New README.md

astormnewrelic opened the pull request:
## Proposed Release Notes

- Streamlined README.md

## Links

- https://github.com/newrelic/node-newrelic/issues/399

## Details

Streamlines README.md by converting it to the New Standard New Relic format, and replacing previous documentation with links to appropriate docs.newrelic.com site.

michaelgoin commented on README.md:
```diff
@@ -1,745 +1,140 @@
+- [New Relic's official NodeJS Agent Documentation](https://docs.newrelic.com/docs/agents/nodejs-agent)
 
-#### newrelic.noticeError(error, customParameters)
+- [Developer Docs](http://newrelic.github.io/node-newrelic/docs/)
```
Not entirely clear this contains the instrumentation API documentation but maybe could be inferred.

Maybe something like "Developer Docs and Instrumentation API".

carlonewrelic commented on README.md:
```diff
@@ -1,745 +1,148 @@
-* thread profiling
-* X-ray transactions (depends on thread profiling)
-* capacity planning
+The NodeJS Agent also uses source code from third-party libraries. You can find full details on which libraries are used and the terms under which they are licensed in [the third-party notices document](/node-newrelic/blob/main/THIRD_PARTY_NOTICES.md).
```
Third-party notices link appears broken in `rich diff` view.  This is the generated link:  `https://github.com/newrelic/node-newrelic/blob/main/node-newrelic/blob/main/THIRD_PARTY_NOTICES.md`
I think the path to the file should be relative to this one so this should work:  `THIRD_PARTY_NOTICES.md`

carlonewrelic commented on README.md:
````diff
@@ -1,745 +1,148 @@
-```js
-newrelic.addCustomAttributes({test: 'value', test2: 'value2'});
-```
+- [The Changelog](/node-newrelic/blob/main/NEWS.md)
````
This link does not work in `rich diff` view.  I believe it should just be `News.md`

astormnewrelic commented:
Changed things up to used language based fenced code blocks (`"```js"` style) and changed links to use fully qualified domain names and avoid all the weirdness with `href`s in different context in GItHub repos.
